### PR TITLE
feat: load clusterfile from template

### PR DESCRIPTION
**Pull request by fengxsong**
Signed-off-by: fengxsong <fengxsong@outlook.com>

https://github.com/labring/sealos/issues/1480

### CODE REVIEW

**Comment by fengxsong** [comment]
More details need to be discussed.

File: pkg/apply/args.go [hunk 1]
```diff
@@ -37,6 +37,12 @@ type RunArgs struct {
 	CustomCMD []string
 }
 
+type ApplyArgs struct {
```
Comment (fanux) [inline pkg/apply/args.go:40]: ApplyArgs->Args

File: pkg/clusterfile/clusterfile.go [hunk 2]
```diff
@@ -55,6 +58,32 @@ func (c *ClusterFile) GetKubeadmConfig() *runtime.KubeadmConfig {
 	return c.KubeConfig
 }
 
-func NewClusterFile(path string) Interface {
-	return &ClusterFile{path: path}
+type ClusterFileOption func(*ClusterFile)
```
Comment (fanux) [inline pkg/clusterfile/clusterfile.go:61]: ClusterFileOption -> Option

File: pkg/apply/args.go (hunk 1 above)
Comment (fengxsong) [reply pkg/apply/args.go:40]: LoL, I was wondering just adding comment `//revive:disable:exported`

File: pkg/clusterfile/clusterfile.go (hunk 2 above)
Comment (fengxsong) [reply pkg/clusterfile/clusterfile.go]: `OptionFunc` more accurate

**Comment by fanux** [comment]
Add docs for this feature

**Review by fanux — APPROVED** [review]

**Pull request merged**